metacat-connector-druid/src/main/java/com.netflix.metacat.connector.druid/DruidConnectorTableService.java

```diff
@@ -0,0 +1,114 @@
+/*
+ *  Copyright 2017 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+
+package com.netflix.metacat.connector.druid;
+
+import com.google.common.collect.Lists;
+import com.netflix.metacat.common.QualifiedName;
+import com.netflix.metacat.common.dto.Pageable;
+import com.netflix.metacat.common.dto.Sort;
+import com.netflix.metacat.common.server.connectors.ConnectorRequestContext;
+import com.netflix.metacat.common.server.connectors.ConnectorTableService;
+import com.netflix.metacat.common.server.connectors.ConnectorUtils;
+import com.netflix.metacat.common.server.connectors.exception.ConnectorException;
+import com.netflix.metacat.common.server.connectors.model.FieldInfo;
+import com.netflix.metacat.common.server.connectors.model.TableInfo;
+import com.netflix.metacat.common.type.BaseType;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Druid Connector Table Service.
+ *
+ * @author zhenl
+ * @since 1.1.0
+ */
+@Slf4j
+public class DruidConnectorTableService implements ConnectorTableService {
+    private final IMetacatDruidClient druidClient;
+
+    /**
+     * Constructor.
+     *
+     * @param druidClient druid client
+     */
+    public DruidConnectorTableService(
+        final IMetacatDruidClient druidClient
+    ) {
+        this.druidClient = druidClient;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public TableInfo get(@Nonnull final ConnectorRequestContext context, @Nonnull final QualifiedName name) {
+        log.debug("Beginning to get table metadata for qualified name {} for request {}", name, context);
+        final Map<String, String> segInfo = this.druidClient.getDataSourceByName(name.getTableName());
```

Comment:
How do we plan to store segment information?

Reply:
Don't plan to store them. We will get them from druid on the fly.